presto-tpch/src/main/java/io/prestosql/plugin/tpch/TpchMetadata.java

```diff
@@ -471,6 +402,89 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
         return builder.build();
     }
 
+    @Override
+    public boolean usesLegacyTableLayouts()
+    {
+        return false;
+    }
+
+    @Override
+    public ConnectorTableProperties getTableProperties(ConnectorSession session, ConnectorTableHandle table)
+    {
+        TpchTableHandle tableHandle = (TpchTableHandle) table;
+
+        Optional<ConnectorTablePartitioning> tablePartitioning = Optional.empty();
+        Optional<Set<ColumnHandle>> partitioningColumns = Optional.empty();
+        List<LocalProperty<ColumnHandle>> localProperties = ImmutableList.of();
+
+        Map<String, ColumnHandle> columns = getColumnHandles(session, tableHandle);
+        if (partitioningEnabled && tableHandle.getTableName().equals(TpchTable.ORDERS.getTableName())) {
+            ColumnHandle orderKeyColumn = columns.get(columnNaming.getName(OrderColumn.ORDER_KEY));
+            tablePartitioning = Optional.of(new ConnectorTablePartitioning(
+                    new TpchPartitioningHandle(
+                            TpchTable.ORDERS.getTableName(),
+                            calculateTotalRows(OrderGenerator.SCALE_BASE, tableHandle.getScaleFactor())),
+                    ImmutableList.of(orderKeyColumn)));
+            partitioningColumns = Optional.of(ImmutableSet.of(orderKeyColumn));
+            localProperties = ImmutableList.of(new SortingProperty<>(orderKeyColumn, SortOrder.ASC_NULLS_FIRST));
+        }
+        else if (partitioningEnabled && tableHandle.getTableName().equals(TpchTable.LINE_ITEM.getTableName())) {
+            ColumnHandle orderKeyColumn = columns.get(columnNaming.getName(LineItemColumn.ORDER_KEY));
+            tablePartitioning = Optional.of(new ConnectorTablePartitioning(
+                    new TpchPartitioningHandle(
+                            TpchTable.ORDERS.getTableName(),
+                            calculateTotalRows(OrderGenerator.SCALE_BASE, tableHandle.getScaleFactor())),
+                    ImmutableList.of(orderKeyColumn)));
+            partitioningColumns = Optional.of(ImmutableSet.of(orderKeyColumn));
+            localProperties = ImmutableList.of(
+                    new SortingProperty<>(orderKeyColumn, SortOrder.ASC_NULLS_FIRST),
+                    new SortingProperty<>(columns.get(columnNaming.getName(LineItemColumn.LINE_NUMBER)), SortOrder.ASC_NULLS_FIRST));
+        }
+
+        return new ConnectorTableProperties(
+                tableHandle.getConstraint(),
+                tablePartitioning,
+                partitioningColumns,
+                Optional.empty(),
+                localProperties);
+    }
+
+    public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(ConnectorSession session, ConnectorTableHandle table, Constraint constraint)
+    {
+        TpchTableHandle handle = (TpchTableHandle) table;
+
+        TupleDomain<ColumnHandle> oldDomain = handle.getConstraint();
```

Comment:
The old domain is being lost in the new TableHandle. If it's not going to be returned as part of the unenforced constraint, it needs to be "remembered". 

This happens to work for now because PushPredicateIntoTableScan keeps track of the predicate in `TableScanNode#enforcedConstraint`, but that will be removed when TableLayouts are finally gone.

A query like this would reproduce the issue:

```sql
SELECT * FROM (
    SELECT * FROM orders WHERE orderstatus = 'O'
)
WHERE orderstatus in ('O', 'F')
```